aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -108,8 +108,7 @@ module aptos_framework::object {
     /// This is a one time ability given to the creator to configure the object as necessary
     struct ConstructorRef has drop {
         self: address,
-        /// Set to true so long as deleting the object is possible. For example, the object was
-        /// created via create_object_from_guid.
+        /// Set to true so long as deleting the object is possible. 
```

Comment:
add to 53rd line of this file:

// transaction_context::create_unique_address uses this for domain separation within it's native implementation
#[test_only]
const DERIVE_UUID_ADDRESS_SCHEME: u8 = 0xFB;


and add unit test that checks that aptos_framework::transaction_context::create_unique_address() returns the same as if you took 
        let guid = Guid(aptos_framework::transaction_context::get_txn_hash(), 0 /* because 0 is the first sequence number I assume */)
        let bytes = bcs::to_bytes(&guid);
        vector::push_back(&mut bytes, DERIVE_UUID_ADDRESS_SCHEME);
        let obj_addr = from_bcs::to_address(hash::sha3_256(bytes));


to make sure that serialization, hashing and domain separation works the same way and correctly

Reply:
I added the test case to `transaction_context.move` instead of `object.move`. Is that okay?